Rust/src/conditionalfieldsummarizer.rs

```diff
@@ -0,0 +1,78 @@
+
+use num::abs;
+use crate::pointstore::PointStore;
+use crate::samplesummary::{SampleSummary, summarize};
+
+fn project_missing(point: &Vec<f32>,position : &[usize]) -> Vec<f32> {
+    position.iter().map(|i| point[*i]).collect()
+}
+
+pub fn field_summarizer(pointstore: &dyn PointStore, point_list_with_distance : &[(usize, f32)], missing: &[usize], centrality: f64, project: bool, max_number : usize) -> SampleSummary {
```

Comment:
This is a generic function name that takes a bunch of arguments. Can we add some documentation?

Should this be a trait?
```
pub trait FieldSummarizer {
  fn summarize(...);
  fn project_missing(...);
}
```

Reply:
Field summarizer is just supposed to be a fn, so a bit simpler than a trait? The project missing is a simple function that projects down to the m missing fields in impute -- however there are use cases where the full projection is also beneficial. Updating with documentation and hopefully some tests!